### PR TITLE
Fix var_export on ContainerConfigurator

### PR DESCRIPTION
**Pull request by maxhelias**
Fix var_export on ContainerConfigurator

### CODE REVIEW

File: src/Configurator/ContainerConfigurator.php [hunk 1]
```diff
@@ -76,13 +76,31 @@ private function addParameters(array $parameters)
...
+            $parametersLines[] = sprintf("    %s: '%s'%s", $key, $value, "\n");
         }
         if (!$endAt) {
             $parametersLines[] = "\n";
         }
         array_splice($lines, $endAt, 0, $parametersLines);
         file_put_contents($target, implode('', $lines));
     }
+
+    private function parse($level, $array): string
```
Comment (nicolas-grekas) [inline src/Configurator/ContainerConfigurator.php]: `dumpYaml()`? could deserve some test cases?

File: src/Downloader.php [hunk 2]
```diff
@@ -110,8 +110,7 @@ public function getRecipes(array $operations): array
                 }
             }
 
-            // FIXME: getNames() can return n names
-            $name = str_replace('/', ',', $package->getNames()[0]);
+            $name = str_replace('/', ',', $package->getName());
```
Comment (nicolas-grekas) [inline src/Downloader.php]: This look suspicious to me: there must be a reason why we didn't write this code originally.
What should we do with the extra names? When does it happen in the first place?
This should be investigated to me.

Comment (maxhelias) [reply src/Downloader.php]: I agree, but I try to find out why and I still do not have the answer. `getNames()` always return `getName()` first :
https://github.com/composer/composer/blob/master/src/Composer/Package/BasePackage.php#L95

File: src/Configurator/ContainerConfigurator.php [hunk 3]
```diff
@@ -76,13 +76,31 @@ private function addParameters(array $parameters)
             $parametersLines[] = "parameters:\n";
         }
         foreach ($parameters as $key => $value) {
-            // FIXME: var_export() only works for basics types, but we don't have access to the Symfony YAML component here
-            $parametersLines[] = sprintf('    %s: %s%s', $key, var_export($value, true), "\n");
+            if (\is_array($value)) {
+                $parametersLines[] = $this->parse(1, $value);
+                continue;
+            }
+            $parametersLines[] = sprintf("    %s: '%s'%s", $key, $value, "\n");
```
Comment (fabpot) [inline src/Configurator/ContainerConfigurator.php]: What if `$value` contains `'`?

File: src/Downloader.php (hunk 2 above)
Comment (fabpot) [reply src/Downloader.php]: Works for me. The FIXME was more about what to do when we have more names. So, I would keep the FIXME and change the currently implementation as you did with `getName()`.

File: src/Configurator/ContainerConfigurator.php (hunk 3 above)
Comment (maxhelias) [reply src/Configurator/ContainerConfigurator.php]: I add `addslashes` for escape `$value`

File: src/Configurator/ContainerConfigurator.php (hunk 1 above)
Comment (maxhelias) [reply src/Configurator/ContainerConfigurator.php]: There was still some bug that are now corrected and covered by tests 😃

File: src/Configurator/ContainerConfigurator.php [hunk 4]
```diff
@@ -76,13 +74,45 @@ private function addParameters(array $parameters)
             $parametersLines[] = "parameters:\n";
         }
         foreach ($parameters as $key => $value) {
-            // FIXME: var_export() only works for basics types, but we don't have access to the Symfony YAML component here
-            $parametersLines[] = sprintf('    %s: %s%s', $key, var_export($value, true), "\n");
+            if (\is_array($value)) {
+                $parametersLines[] = sprintf("    %s:\n%s", $key, $this->dumpYaml(2, $value));
+                continue;
+            }
+            $parametersLines[] = sprintf("    %s: '%s'%s", $key, addslashes($value), "\n");
```
Comment (fabpot) [inline src/Configurator/ContainerConfigurator.php]: What about using str_replace instead as we only need to escape quotes?

Comment (maxhelias) [reply src/Configurator/ContainerConfigurator.php]: Yes, I think it will be better because i juste thinking it's not a slash for escape in yaml

**Comment by fabpot** [comment]
Thank you @maxhelias.

**Pull request merged**